src/rasterstats/main.py

```diff
@@ -302,6 +302,9 @@ def gen_zonal_stats(
                 feature_stats = prefixed_feature_stats
 
             if geojson_out:
+                if not isinstance(feat, dict):
+                    feat = dict(feat, type=feat.__class__.__name__)
+                    feat["properties"] = dict(feat["properties"])
```

Comment:
If I understand this correctly, `feat` will be type `dict` for `fiona <=1.8` and type `fiona.model.Feature` for 1.9+? I don't think that's the correct solution for a few reasons:

- Dealing with two different potential types effectively doubles the API debugging surface area. At the very minimum, we'd need to extend the test suite to run under both versions of fiona to ensure full coverage. Additionally, unit testing should be done by passing the different types explicitly.
-  Ideally we could handle this entirely in `io.py` to keep it clean and testable; normalize all `Features` to `dicts` and have a test case for it.
- There's probably a faster way to do this - I want to be very cautious about causing a performance regression for anyone on Fiona 1.9+. Again, we should probably encapsulate the serialization logic in `io.py` so we can test/benchmark it explicitly.

Reply:
Aha, [`fiona.model.to_dict`](https://github.com/Toblerity/Fiona/blob/eaf6e0a5948b2b7585b7b216c90e8704e1d88904/fiona/model.py#L413-L420) is "the way" to convert Fiona 1.9+ objects to dict. I'll have another look to see how to abstract this better in `io.py`. Agree that testing with Fiona<1.9 should be done for perhaps another couple of years.